tools/find_tap_name/main.go

```diff
@@ -0,0 +1,143 @@
+  // netConfigKeyPath is the registry location of the network adapters network configuration.
+  netConfigKeyPath = `SYSTEM\CurrentControlSet\Control\Network\{4D36E972-E325-11CE-BFC1-08002BE10318}`
+)
+
+// getAdapterNameAndInstallTimestamp returns the name and install timestamp of a network adapter with
+// registry location `adapterKeyPath`. Returns a non-nil error on failure, or if the adapter's
+// hardware component ID does not match `componentID`.
+func getAdapterNameAndInstallTimestamp(adapterKeyPath, componentID string) (name string, installTimestamp uint64, err error) {
+  adapterKey, err := registry.OpenKey(registry.LOCAL_MACHINE, adapterKeyPath, registry.READ)
+  if err != nil {
+    log.Println("Failed to open adapter key:", err)
+    return
+  }
+  defer adapterKey.Close()
+
+  adapterComponentID, _, err := adapterKey.GetStringValue("ComponentId")
+  if err != nil {
+    log.Println("Failed to read adapter component ID:", err)
+    return
+  }
+  log.Println("Found", adapterComponentID)
+  if adapterComponentID != componentID {
+    err = fmt.Errorf("Network adapter component ID does not match %v", componentID)
+    return
+  }
+
+  installTimestampBytes, _, err := adapterKey.GetBinaryValue("InstallTimeStamp")
+  if err != nil {
+    log.Println("Failed to read adapter install timestamp:", err)
+    return
+  }
+  // Although Windows is little endian, we have observed that network adapters' install timestamps
+  // are encoded as big endian in the registry.
+  installTimestamp = binary.BigEndian.Uint64(installTimestampBytes)
+  log.Println("\tInstall timestamp", installTimestamp)
+
+  netConfigID, _, err := adapterKey.GetStringValue("NetCfgInstanceId")
+  if err != nil {
+    log.Println("Failed to read network configuration ID:", err)
+    return
+  }
+  adapterConfigKeyPath := fmt.Sprintf("%s\\%s\\Connection", netConfigKeyPath, netConfigID)
+  adapterConfigKey, err := registry.OpenKey(registry.LOCAL_MACHINE, adapterConfigKeyPath, registry.READ)
+  if err != nil {
+    log.Println("Failed to open network configuration key:", err)
+    return
+  }
+  defer adapterConfigKey.Close()
+
+  name, _, err = adapterConfigKey.GetStringValue("Name")
+  if err != nil {
+    log.Println("Failed to read adapter name:", err)
+    return
+  }
+  log.Println("\tName", name)
+  return
+}
+
+// findNetworkAdapterName searches the Windows registry for the name of a network adapter with
+// `componentID`. Since there may be more than one network adapter with the same component ID,
+// selects the most recently installed device in the event of a conflict.
+// Returns an empty string and an error if the device name cannot be found.
+func findNetworkAdapterName(componentID string) (string, error) {
+  netAdaptersKey, err := registry.OpenKey(registry.LOCAL_MACHINE, netAdaptersKeyPath, registry.READ)
+  if err != nil {
+    return "", fmt.Errorf("Failed to open the network adapter registry, %v", err)
```

Comment:
Nit: This seems like a good case for the new `%w` option.

Reply:
Done.